shlib/shcrypto/encryption.go

```diff
@@ -60,8 +60,7 @@ func Encrypt(message []byte, eonPublicKey *EonPublicKey, epochID *EpochID, sigma
 }
 
 func computeR(sigma Block, message []byte) *big.Int {
-	messageHash := HashBytesToBlock(message)
-	return HashBlocksToInt(sigma, messageHash)
+	return Hash3(append(sigma[:], message...))
```

Comment:
I wasn't sure and did some research on this, so I just wanted to share what I found:

Be aware that the append only goes well because you create a slice covering the full underlying array `[:]`.
If for whatever reason, you would do `append(sigma[:30], message...))` and `len(message)==2` you would overwrite the last 2 bytes of the underlying sigma array. This is because `cap(sigma) <= len(sigma[:30]) + len(message)`.

(for more info see this [go-playground](https://go.dev/play/p/ZRB9w6uAmg5) and [blog](https://go.dev/blog/slices-intro))